lib/jsonld-signatures.js

```diff
@@ -151,6 +151,10 @@ api.sign = function(input, options, callback) {
   var nonce = options.nonce || null;
   var algorithm = options.algorithm || 'GraphSignature2012';
 
+  // save any previous signatures for later
+  var previousSignature = input.signature || null;
+  delete input.signature;
```

Comment:
This processing should happen after the `normalize()` call:
- The input might be using some alias other than "signature" for the signature property (or even no alias).
- It would be better to leave the input unmodified and do manipulations on the new object that `normalize()` creates.